### DISCUSSION TITLE
Azure Function Connect-PnpOnline ManagedIdentity ERROR: Host not reachable

### DISCUSSION THREAD

**Discussion by jonathanfurrer** [pnp/powershell#3151] [Q&A]
I tried to connect inside an Azure Function with Managed Identity (System Assigned). 
I followed [this ](https://pnp.github.io/powershell/articles/azurefunctions.html)
PowerShell Version = 7.2
PnP.PowerShell = 2.1.65-nightly
Runtime version = 4.21.2.20699

$siteUrl = "https://tenant.sharepoint.com/sites/testsite"
Connect-PnPOnline -Url $siteUrl -ManagedIdentity

I get this error. Anyone an idea? 

  ```
2023-06-02T13:32:28Z   [Error]   ERROR: Host not reachable
  
  Exception             : 
      Type        : System.Management.Automation.PSArgumentException
      ErrorRecord : 
          Exception             : 
              Type    : System.Management.Automation.ParentContainsErrorRecordException
              Message : Host not reachable
              HResult : -2146233087
          CategoryInfo          : InvalidArgument: (:) [], ParentContainsErrorRecordException
          FullyQualifiedErrorId : Argument
      Message     : Host not reachable
      TargetSite  : 
          Name          : Connect
          DeclaringType : PnP.PowerShell.Commands.Base.ConnectOnline
          MemberType    : Method
          Module        : PnP.PowerShell.dll
      Data        : System.Collections.ListDictionaryInternal
      Source      : PnP.PowerShell
      HResult     : -2147024809
      StackTrace  : 
     at PnP.PowerShell.Commands.Base.ConnectOnline.Connect(CancellationToken& cancellationToken) in D:\a\powershell\powershell\src\Commands\Base\ConnectOnline.cs:line 282
     at PnP.PowerShell.Commands.Base.ConnectOnline.ProcessRecord() in D:\a\powershell\powershell\src\Commands\Base\ConnectOnline.cs:line 248
     at System.Management.Automation.CommandProcessor.ProcessRecord()
  CategoryInfo          : InvalidArgument: (:) [Connect-PnPOnline], PSArgumentException
  FullyQualifiedErrorId : Argument,PnP.PowerShell.Commands.Base.ConnectOnline
  InvocationInfo        : 
      MyCommand        : Connect-PnPOnline
      ScriptLineNumber : 15
      OffsetInLine     : 1
      HistoryId        : 1
      ScriptName       : C:\home\site\wwwroot\powerAutomatTrigger\run.ps1
      Line             : Connect-PnPOnline -Url $siteUrl -ManagedIdentity
                         
      PositionMessage  : At C:\home\site\wwwroot\powerAutomatTrigger\run.ps1:15 char:1
                         + Connect-PnPOnline -Url $siteUrl -ManagedIdentity
                         + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      PSScriptRoot     : C:\home\site\wwwroot\powerAutomatTrigger
      PSCommandPath    : C:\home\site\wwwroot\powerAutomatTrigger\run.ps1
      InvocationName   : Connect-PnPOnline
      CommandOrigin    : Internal
  ScriptStackTrace      : at <ScriptBlock>, C:\home\site\wwwroot\powerAutomatTrigger\run.ps1: line 15
```

**Comment by stefanschoemaker** [comment]
I just had the same problem, use an older version of the module. 

The current version in https://github.com/pnp/powershell/blob/cc3c5564fca9ce96b1a2ac47c7aabdc8b90136d0/src/Commands/Base/ConnectOnline.cs#L282
uses 

`            if (PingHost(new Uri(Url).Host) == false)
            {
                throw new PSArgumentException("Host not reachable");
            }
`

Ping is not allowed through an Azure function and errors out because of this.

**Reply by jonathanfurrer** [reply]
The 2.1.53-nightly build works. Thx

**Reply by gautamdsheth** [reply]
Thanks for testing out the nightly builds.
Have fixed the issue related to this ping, it will be available in tomorrow's nightly builds, 2.1.67 or later versions

**Reply by YuriySamorodov** [reply]
Hi @gautamdsheth , got the same error with `PowerShell 7.4.0-preview3 `and `PnP.Powershell 2.1.81` on Mac

**Reply by gautamdsheth** [reply]
@veronicageek - can you please try to repro as well on a Mac , I can't repro it here in Windows ?

**Reply by deanexperian** [reply]
Just wanted to follow up to see if there is any resolution to this?  I am having the same issue with 2.2.  Thank you!

**Reply by wwcc-jack-carico** [reply]
Having this issue currently. The Connect-PnPOnline cmdlet is wowrking on my windows 11 development environment. however when trying to deploy to a production environment I am getting very inconsistent behavior. I will go into detail on the Server 2022 environment but it seems to be happening under server 2016 as well. 

Starting with a fresh deployment of Server 2022 prepared by our server team. I installed powershell 7 x64 build (PowerShell-7.3.5-win-x64.msi)

PS C:\TaskScheduler\SmartsheetsData> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.3.5
PSEdition                      Core
GitCommitId                    7.3.5
OS                             Microsoft Windows 10.0.20348
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

From an administrative pwsh.exe terminal I installed the PnP.Powershell Module.  I also installed the same powershell module using an unprivileged account we intend to use for task automation. 
PS C:\TaskScheduler\SmartsheetsData> Get-InstalledModule

Version              Name                                Repository           Description
-------              ----                                ----------           -----------
2.2.0                PnP.PowerShell                      PSGallery            Microsoft 365 Patterns and Practices PowerShell Cmdlets
22.1.1               SqlServer                           PSGallery            This module allows SQL Server developers, administrators and business intelligence professio…

Currently the Connect-PnpOnline works from a terminal (not elevated) using my admin account identity. 
However when trying to run the same script from a task scheduler instance using the same admin identity the script logs the same error others are reporting "Connect-PnPOnline: Host not reachable". It also logs this error when trying to run the same script using the unprivileged account from a terminal window. 

We are using this form of the command: 
Connect-PnPOnline -Url  [our SP site] -ClientId [app registration client id] -CertificatePath [cert file with private key] -Tenant [our tennant onmicrosoft.com]

I said inconsistent because I have also experienced it where even under the admin account it returns the "Connect-PnPOnline: Host not reachable" response.

**Comment by ChristopherFoerster** [comment]
Is there a reason why this somehow broken feature introduced in 2.2.0 is still present and not yet removed? We are facing the same issue under Windows and Docker that up to 2.1.1 the connection is working and from 2.2.0 we face the error.
